core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -71,6 +72,16 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val actorSyste
   // Executors we have requested the cluster manager to kill that have not died yet
   private val executorsPendingToRemove = new HashSet[String]
 
+  /**
+   * Send new credentials to executors. This is the method that is called when the scheduled
+   * login completes, so the new credentials can be sent to the executors.
+   * @param credentials
+   */
+  def sendNewCredentialsToExecutors(credentials: SerializableBuffer): Unit = {
+    // We don't care about the reply, so going to deadLetters is fine.
```

Comment:
The message has to reach all executors. But the issue is that if the message does not reach the executors, they don't have a way to ask for it. 

We could change it to a pull based mechanism where an executor asks for "new credentials" -- but that would mean that both the executor and the AM need to have expiration (since the AM would renew credentials when executor asks for it, if it is old). I am open to this approach as well. This is not required if akka's "!" actually guarantees delivery, else we could switch to this one.